src/common.jl

```diff
@@ -72,6 +72,22 @@ nsamples(::Type{D}, x::AbstractMatrix) where {D<:Sampleable{Multivariate}} = siz
 nsamples(::Type{D}, x::Number) where {D<:Sampleable{Matrixvariate}} = 1
 nsamples(::Type{D}, x::Array{Matrix{T}}) where {D<:Sampleable{Matrixvariate},T<:Number} = length(x)
 
+for func in (:(==), :isequal, :isapprox)
+    @eval function Base.$func(s1::A, s2::B; kwargs...) where {A<:Sampleable, B<:Sampleable}
+        nameof(A) == nameof(B) || return false
+        fields = fieldnames(A)
+        fields == fieldnames(B) || return false
```

Comment:
Could help the compiler determine this statically for differing types:
```suggestion
        nameof(A) === nameof(B) || return false
        fields = fieldnames(A)
        fields === fieldnames(B) || return false
```